### ISSUE TITLE
Add ddb delta command

### ISSUE THREAD

**Issue by pierotofy** [DroneDB/DroneDB#174]
- [ ] Implement `ddb delta <dataset A> <dataset B>`

The result should be a list of file operation changes needed to create B from A (addition/move/deletion/rename).

**Comment by pierotofy** [comment]
Useful document: https://rsync.samba.org/how-rsync-works.html

**Comment by pierotofy** [comment]
Also notes on how rsync handles interruptions: https://rsync.samba.org/how-rsync-works.html

**Comment by pierotofy** [comment]
- [ ] Create test cases covering trivial and non-trivial cases

**Comment by pierotofy** [comment]
- [ ] Output in JSON or text

**Comment by HeDo88TH** [comment]
Proposal:

```
"dataset A": ["pippo.jpg", "pluto.jpg", "minnie.jpg", "gambadilegno.jpg", "1.jpg", "pics/2.jpg", "pics/3.jpg"]	
"dataset B": ["pippo.jpg", "pluto.jpg", "minnie.jpg", "minnie2.jpg", "topolino.jpg", "1.jpg", "images/2.jpg", "images/4.jpg"]	
```

`ddb delta <dataset A> <dataset B>`

Output:

```
{
    "source": "elkiuhjrbkehffrg",
    "destination": "loieureriughguieh",
    "date": 145676454365,
    "actions": {
        "add": [
            "topolino.jpg",
            "minnie2.jpg", 
            "images/4.jpg"
        ],
        "remove": [
            "gambadilegno.jpg",
            "pics/3.jpg",
            "pics"
        ],
        "move": [{
                "source": "pics/2.jpg",
                "destination": "images/2.jpg"
            }
        ]
    }
}
```

**Comment by pierotofy** [comment]
Proposed improvement (you don't need source, destination and date, move operations can be binary and thus only use an array).

```
{
        "add": [
            "topolino.jpg",
            "minnie2.jpg", 
            "images/4.jpg"
        ],
        "remove": [
            "gambadilegno.jpg",
            "pics/3.jpg",
            "pics"
        ],
        "move": [
              ["pics/2.jpg", "images/2.jpg"]
        ]
}
```

**Comment by pierotofy** [comment]
Note that filenames might be the same, but refer to different files (e.g. a file has been edited).

**Comment by HeDo88TH** [comment]
Closed by #182
